pkg/fileutils/fileutils.go

```diff
@@ -57,14 +57,34 @@ func NewPatternMatcher(patterns []string) (*PatternMatcher, error) {
 	return pm, nil
 }
 
+type MatchResult struct {
+	isMatched         bool
+	matches, excludes uint
+}
+
+// Excludes returns true if the overall result is matched
+func (m *MatchResult) IsMatched() bool {
+	return m.isMatched
+}
+
+// Excludes returns the amount of matches of an MatchResult
+func (m *MatchResult) Matches() uint {
+	return m.matches
+}
+
+// Excludes returns the amount of excludes of an MatchResult
+func (m *MatchResult) Excludes() uint {
+	return m.excludes
+}
+
 // Matches verifies the provided filepath against all patterns.
-// It returns the amount of `matches` and `excludes` for the patterns on
-// success, otherwise an error.
-// It is not safe to be called concurrently.
-func (pm *PatternMatcher) Matches(file string) (matches, excludes uint, err error) {
+// It returns the `*MatchResult` result for the patterns on success, otherwise
+// an error. This method is not safe to be called concurrently.
+func (pm *PatternMatcher) Matches(file string) (res *MatchResult, err error) {
```

Comment:
Could you depracate the old function and add a  new function rather then modify the existing, so that we don't need to bump Major Version. CountMatches?

Reply:
Hm, I'm a bit unsure [since it seems that the previous PRs regarding changes to this method have not been release yet](https://github.com/containers/storage/compare/v1.13.5...saschagrunert:history-names). Should I revert this method completely back to the version of 1.13.5 and then add another in in parallel?

Reply:
Yes.

Reply:
Done.